packages/ui-img/src/Img/index.js

```diff
@@ -21,213 +21,156 @@
+import { ThemeablePropTypes } from '@instructure/ui-themeable'
+import { passthroughProps } from '@instructure/ui-react-utils'
 import { supportsObjectFit } from '@instructure/ui-dom-utils'
-import { testable } from '@instructure/ui-testable'
 
-import styles from './styles.css'
-import theme from './theme'
+import generateStyle from './styles'
 
 /**
 ---
 category: components
 ---
 **/
-@deprecated('8.0.0', {
-  grayscale: 'withGrayscale',
-  blur: 'withBlur',
-  inline: 'display'
-})
-@testable()
-@themeable(theme, styles)
-class Img extends Component {
-  static propTypes = {
-    src: PropTypes.string.isRequired,
-    alt: PropTypes.string,
-    display: PropTypes.oneOf(['inline-block', 'block']),
-    /**
-     * Valid values are `0`, `none`, `auto`, `xxx-small`, `xx-small`, `x-small`,
-     * `small`, `medium`, `large`, `x-large`, `xx-large`. Apply these values via
-     * familiar CSS-like shorthand. For example: `margin="small auto large"`.
-     */
-    margin: ThemeablePropTypes.spacing,
-    /**
-     * Valid values for `opacity` are `0` - `10`. Valid values for `blend` are
-     * `normal` (default), `multiply`, `screen`, `overlay`, and `color-burn`.
-     */
-    overlay: PropTypes.shape({
-      color: PropTypes.string.isRequired,
-      opacity: PropTypes.oneOf([0, 1, 2, 3, 4, 5, 6, 7, 8, 9, 10]).isRequired,
-      blend: PropTypes.oneOf([
-        'normal',
-        'multiply',
-        'screen',
-        'overlay',
-        'color-burn'
-      ])
-    }),
-    withGrayscale: PropTypes.bool,
-    withBlur: PropTypes.bool,
-    constrain: PropTypes.oneOf(['cover', 'contain']),
-    elementRef: PropTypes.func,
-    height: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
-    width: PropTypes.oneOfType([PropTypes.string, PropTypes.number]),
-    /* eslint-disable react/require-default-props */
-    /**
-     * __Deprecated - use `display`__
-     */
-    inline: PropTypes.bool,
-    /**
-     * __Deprecated - use `withGrayscale`__
-     */
-    grayscale: PropTypes.bool,
-    /**
-     * __Deprecated - use `withBlur`__
-     */
-    blur: PropTypes.bool
-    /* eslint-enable react/require-default-props */
-  }
+const Img = (props) => {
+  const {
+    src,
+    alt,
+    margin,
+    display,
+    overlay,
+    withGrayscale,
+    withBlur,
+    constrain,
+    width,
+    height,
+    elementRef,
+    ...restProps
+  } = props
 
-  static defaultProps = {
-    margin: undefined,
-    overlay: undefined,
-    constrain: undefined,
-    elementRef: undefined,
-    height: undefined,
-    width: undefined,
-    alt: '',
-    display: 'inline-block',
-    withGrayscale: false,
-    withBlur: false
+  const a11yProps = {
+    alt: alt || ''
   }
 
-  get supportsObjectFit() {
-    return supportsObjectFit()
-  }
+  // if browser does not support ObjectFit CSS, and Img needs "constrain",
+  // serve up a background-image instead
+  const hasBackground = !supportsObjectFit() && props.constrain
```

Comment:
style logic like these could be moved to styles.js, we can use props/state/imports there as well